### ISSUE TITLE
Will they support MAUI?

### ISSUE THREAD

**Issue by Beeron2010** [CrossGeeks/FirebasePushNotificationPlugin#408]
Hi there,

I know that you don't officially support MAUI at the moment.
However, MAUI should appear in Q2 in 2022.

Will they offer support?
Unfortunately, errors are currently occurring and we are starting to use MAUI on our projects so that we only have to update it later.

Currently joining VS2022, NET 6 (MAUI Preview 12)
the following error:

---1---
Severity Code Description Project File Line Suppression state
Error XA4215 `mono.android.support.v4.media.session.MediaSessionCompat_OnActiveChangeListenerImplementor` generated by: Android.Support.V4.Media.Session.MediaSessionCompat+IOnActiveChangeListenerImplementor, Xamarin.Android.Support.Media.Compat, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null Bevie-Times C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\31.0.200-preview.12.2\tools\Xamarin.Android.Common.targets 1429

--- 2 ---
Severity Code Description Project File Line Suppression state
Error XA4215 `mono.android.support.v4.media.session.MediaSessionCompat_OnActiveChangeListenerImplementor` generated by: Android.Support.V4.Media.Session.MediaSessionCompat+IOnActiveChangeListenerImplementor, Xamarin.AndroidX.Media, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null Bevie-Times C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\31.0.200-preview.12.2\tools\Xamarin.Android.Common.targets 1429

--- 3 ---

Severity Code Description Project File Line Suppression state
Error XA4215 The Java type `mono.android.support.v4.media.session.MediaSessionCompat_OnActiveChangeListenerImplementor` is generated by more than one managed type. Please change the [Register] attribute so that the same Java type is not emitted. Bevie-Times C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\31.0.200-preview.12.2\tools\Xamarin.Android.Common.targets 1429

Many Thanks ;)

**Comment by JeffIPvideo** [comment]
pre-release is also coming this quarter
https://github.com/dotnet/maui/wiki/roadmap.
For the most part the latest release preview 12 is in its final form.
Support for MAUI will be great appreciated

**Comment by tvkit** [comment]
This project relies on Xamarin.Firebase.iOS.* which doesn't yet support net6.0.  

https://github.com/xamarin/GoogleApisForiOSComponents

**Comment by Beeron2010** [comment]
@tvkit Yeah but i need only Android for the Moment. And the Notification Hub from Azure is for Simple Messages to Expensive

**Comment by tvkit** [comment]
> @tvkit Yeah but i need only Android for the Moment. And the Notification Hub from Azure is for Simple Messages to Expensive

@Beeron2010  it works for Android.  You can include the require packages in a conditional android group

```
    <ItemGroup Condition="$(TargetFramework.Contains('-android'))">
        <PackageReference Include="..." />
    </ItemGroup>
```

**Comment by Hantse** [comment]
Some news about MAUI version ?

**Comment by JeffIPV** [comment]
> Some news about MAUI version ?

This project relies on Xamarin.Firebase.iOS.* which doesn't yet support net6.0.
They have been marked ready for a while but not released. It seems it is not if importance to them since Firebase is a competing product to Azure Notifications

https://github.com/xamarin/GoogleApisForiOSComponents/issues/518
https://github.com/xamarin/GoogleApisForiOSComponents/issues/519

I would recommend setting up for APNS directly. You can keep firebase for android.
I can confirm maui works with APNS and I get notifications

**Comment by yepeekai** [comment]
Seen in the notes : Includes support for .net6.0-ios
[https://github.com/xamarin/GoogleApisForiOSComponents/releases/tag/Firebase-8.10.0.1](https://github.com/xamarin/GoogleApisForiOSComponents/releases/tag/Firebase-8.10.0.1)

So... will this project support this?

**Comment by kylewhittington** [comment]
Any update on this? Currently trying to get Firebase push to work on MAUI without much luck. Just attempting Android at the moment.

**Comment by GAURI02X** [comment]
any update on MAUI compatibility with MAUI. Cannot use APNS directly.

**Comment by sstahurski** [comment]
This does not work with MAUI IOS

**Comment by sdavis72** [comment]
I am still getting this issue and don't have a resolution.  Can anyone help me?

1>Done building project "LTS Mobile Maui.csproj".
1>C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\33.0.4\tools\Xamarin.Android.Common.targets(1476,3): warning XA4214: The managed type `Android.Support.V4.Media.Session.MediaSessionCompat.IOnActiveChangeListenerImplementor` exists in multiple assemblies: Xamarin.Android.Support.Media.Compat, Xamarin.AndroidX.Media. Please refactor the managed type names in these assemblies so that they are not identical.
1>C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\33.0.4\tools\Xamarin.Android.Common.targets(1476,3): warning XA4214: References to the type `Android.Support.V4.Media.Session.MediaSessionCompat.IOnActiveChangeListenerImplementor` will refer to `Android.Support.V4.Media.Session.MediaSessionCompat.IOnActiveChangeListenerImplementor, Xamarin.Android.Support.Media.Compat`.
1>C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\33.0.4\tools\Xamarin.Android.Common.targets(1476,3): error XA4215: The Java type `mono.android.support.v4.media.session.MediaSessionCompat_OnActiveChangeListenerImplementor` is generated by more than one managed type. Please change the [Register] attribute so that the same Java type is not emitted.
1>C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\33.0.4\tools\Xamarin.Android.Common.targets(1476,3): error XA4215:   `mono.android.support.v4.media.session.MediaSessionCompat_OnActiveChangeListenerImplementor` generated by: Android.Support.V4.Media.Session.MediaSessionCompat+IOnActiveChangeListenerImplementor, Xamarin.Android.Support.Media.Compat, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
1>C:\Program Files\dotnet\packs\Microsoft.Android.Sdk.Windows\33.0.4\tools\Xamarin.Android.Common.targets(1476,3): error XA4215:   `mono.android.support.v4.media.session.MediaSessionCompat_OnActiveChangeListenerImplementor` generated by: Android.Support.V4.Media.Session.MediaSessionCompat+IOnActiveChangeListenerImplementor, Xamarin.AndroidX.Media, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null
1>Done building project "LTS Mobile Maui.csproj" -- FAILED.
========== Rebuild All: 0 succeeded, 1 failed, 0 skipped ==========
========== Elapsed 00:27.130 ==========

**Comment by Lavankumarkalimisetty** [comment]
Any Update on MAUI? which supports for .net 7.0?

**Comment by thomasgalliker** [comment]
Maybe you‘ve seen it already, if not, try this: [Plugin.FirebasePushNotifications](https://www.nuget.org/packages/Plugin.FirebasePushNotifications)
